wandb/sdk/launch/_project_spec.py

```diff
@@ -54,8 +61,20 @@ def __init__(
         if "entry_point" in overrides:
             self.add_entry_point(overrides["entry_point"])
 
+        if utils._is_wandb_uri(self.uri):
+            self.source = LaunchSource.WANDB
+            self.project_dir = tempfile.mkdtemp()
+        elif utils._is_github_uri(self.uri):
+            self.source = LaunchSource.GITHUB
+            self.project_dir = tempfile.mkdtemp()
+        else:
```

Comment:
We should have a regex check to see if it is indeed a local directory, and error out if now